cmd/cmd_renew.go

```diff
@@ -132,6 +137,14 @@ func renewForDomains(ctx *cli.Context, client *lego.Client, certsStorage *Certif
 		}
 	}
 
+	if ctx.Bool("no-random-sleep") {
+		// https://github.com/go-acme/lego/issues/1656
+		// https://github.com/certbot/certbot/blob/284023a1b7672be2bd4018dd7623b3b92197d4b0/certbot/certbot/_internal/renewal.py#L472
+		sleepTime := rand.New(rand.NewSource(time.Now().UnixNano())).Int63n(int64(8 * time.Minute))
+		log.Infof("renewal: random delay of %s", time.Duration(sleepTime))
+		time.Sleep(time.Duration(sleepTime) * time.Second)
```

Comment:
Here happens a lot in the first line. Maybe rewrite this to something like:

```suggestion
		const jitter = 8 * time.Minute
		rndSource := rand.NewSource(time.Now().UnixNano())
		rnd := rand.New(rndSource)
		sleepTime := time.Duration(rnd.Int63n(int64(jitter)))
		
		log.Infof("renewal: random delay of %s", sleepTime)
		time.Sleep(time.Duration(sleepTime))
```

In any case, in `time.Sleep(time.Duration(sleepTime) * time.Second)` the multiplication by 10<sup>9</sup> (`time.Second`) must be removed, otherwise `time.Sleep` might pause the execution for a few hundred years.

Reply:
yes I iterated and forget to remove the multiplication